File: 1주차/성실/겹치는 선분의 길이.java
class Solution {
    public int solution(int[][] lines) {
        int answer = 0;
        int[] result = new int[200];
        
        for(int i=0; i<3; i++) {
            int a = lines[i][0];
            int b = lines[i][1];
            
            a += 100;
            b += 100;
            
            if (b-1 == a) {
                result[a] += 1;
            } else {
                for(int j=a; j<b; j++) {
                    result[j] += 1;
                }
            }
        }
        
        for (int i : result) {
            if (i >= 2) {
                answer += 1; 
            }
        }
        
        return answer;
    }
}

File: 1주차/선주/겹치는선분의길이.js
function solution(lines) {

    var [a,b,c] = lines;

    let answer = 0;

    //선분 시작점에 따라 정렬

    if(a[0]<b[0]){

        lines = (b[0]<c[0])? [a,b,c] : (c[0]<a[0])? [c,a,b]:[a,c,b]; //abc acb cab

    }else{//bac bca cba

        lines = (c[0]<a[0]) ? [c,b,a] : (b[0]<c[0])? [b,a,c]:[b,c,a];

    }

    //console.log(lines);    

    //겹치지 않는 선분들의 경우 거르기

    if(a[1]<=b[0] && b[1]<=c[0])return 0;

    //겹치는 곳의 길이의 합 구하기
    answer += Math.abs(a[1]-b[0]);
    answer += Math.abs(b[1]-c[0]);
    answer += Math.abs(a[1]-c[0]);
    

    return answer;
};